sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3396,17 +3396,32 @@ pi_result cuda_piEnqueueMemUnmap(pi_queue command_queue, pi_mem memobj,
 pi_result cuda_piextUSMHostAlloc(void **result_ptr, pi_context context,
                                  pi_usm_mem_properties *properties, size_t size,
                                  pi_uint32 alignment) {
+  static constexpr pi_uint32 max_alignment =
+      0x200; // from empirical testing with CUDA 10.2 on a Tesla K40
+
+  // enforce a valid pointer to the allocated memory
   assert(result_ptr != nullptr);
-  assert(context != nullptr);
-  assert(properties == nullptr);
+  // check the the context is valid
+  if (context == nullptr)
```

Comment:
Please, keep asserts. It's expected that the higher level passes right arguments, otherwise it is an internal bug.